src/tpm2_wrap.c

```diff
@@ -172,6 +172,249 @@ int wolfTPM2_Init(WOLFTPM2_DEV* dev, TPM2HalIoCb ioCb, void* userCtx)
     return rc;
 }
 
+#ifdef WOLFTPM2_ENABLE_CSHARP
+WOLFTPM2_DEV *wolfTPM2_SimpleInit(void)
+{
+    WOLFTPM2_DEV *dev = NULL;
+
+    dev = (WOLFTPM2_DEV *) XMALLOC(sizeof(WOLFTPM2_DEV), NULL,
+                                   DYNAMIC_TYPE_TMP_BUFFER);
+    if (dev == NULL) {
+        return NULL;
+    }
+
+    if (wolfTPM2_Init(dev, NULL, NULL) != TPM_RC_SUCCESS) {
+        return NULL;
+    }
+
+    return dev;
+}
+
+int wolfTPM2_SimpleCleanup(WOLFTPM2_DEV *dev)
+{
+    wolfTPM2_Cleanup(dev);
+    XFREE(dev, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM2_KEYBLOB* wolfTPM2_GetEmptyKeyBlob(void)
+{
+    WOLFTPM2_KEYBLOB* blob = NULL;
+
+    blob = (WOLFTPM2_KEYBLOB *) XMALLOC(sizeof(WOLFTPM2_KEYBLOB), NULL,
+                                      DYNAMIC_TYPE_TMP_BUFFER);
+    if (blob == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(blob, 0, sizeof(WOLFTPM2_KEYBLOB));
+    return blob;
+}
+
+int wolfTPM2_SimpleOpenExisting(WOLFTPM2_DEV* dev)
+{
+    return wolfTPM2_OpenExisting(dev, NULL, NULL);
+}
+
+int wolfTPM2_CleanupKeyBlob(WOLFTPM2_KEYBLOB* blob)
+{
+    XFREE(blob, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM_API TPMT_PUBLIC* wolfTPM2_GetEmptyPublicTemplate(void)
+{
+    TPMT_PUBLIC* template = NULL;
+
+    template = (TPMT_PUBLIC *) XMALLOC(sizeof(TPMT_PUBLIC), NULL,
+                                       DYNAMIC_TYPE_TMP_BUFFER);
+    if (template == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(template, 0, sizeof(TPMT_PUBLIC));
+    return template;
+}
+
+WOLFTPM_API int wolfTPM2_CleanupPublicTemplate(TPMT_PUBLIC* template)
+{
+    XFREE(template, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM_API WOLFTPM2_KEY* wolfTPM2_GetEmptyKey(void)
+{
+    WOLFTPM2_KEY* key = NULL;
+
+    key = (WOLFTPM2_KEY *) XMALLOC(sizeof(WOLFTPM2_KEY), NULL,
+                                    DYNAMIC_TYPE_TMP_BUFFER);
+    if (key == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(key, 0, sizeof(WOLFTPM2_KEY));
+    return key;
+}
+
+WOLFTPM_API int wolfTPM2_CleanupKey(WOLFTPM2_KEY* key)
+{
+    XFREE(key, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM2_HANDLE* wolfTPM2_GetHandleRefFromKey(WOLFTPM2_KEY* key) {
```

Comment:
Any public API should check incoming arg. Please put function brace on new line.